common/djangoapps/student/views.py

```diff
@@ -647,21 +671,9 @@ def dashboard(request):
         'ACTIVATION_EMAIL_SUPPORT_LINK', settings.ACTIVATION_EMAIL_SUPPORT_LINK
     ) or settings.SUPPORT_SITE_LINK
 
-    # Let's filter out any courses in an "org" that has been declared to be
-    # in a configuration
-    org_filter_out_set = configuration_helpers.get_all_orgs()
-
-    # Remove current site orgs from the "filter out" list, if applicable.
-    # We want to filter and only show enrollments for courses within
-    # the organizations defined in configuration for the current site.
-    course_org_filter = configuration_helpers.get_current_site_orgs()
-    if course_org_filter:
-        org_filter_out_set = org_filter_out_set - set(course_org_filter)
-
-    # Build our (course, enrollment) list for the user, but ignore any courses that no
-    # longer exist (because the course IDs have changed). Still, we don't delete those
-    # enrollments, because it could have been a data push snafu.
-    course_enrollments = list(get_course_enrollments(user, course_org_filter, org_filter_out_set))
+    # get the org whitelist or the org blacklist for the current site
+    site_org_whitelist, site_org_blacklist = get_org_black_and_whitelist_for_site(user)
```

Comment:
@douglashall: I decided to try to fix this code that I had written a while back to not lose it.

I don't know if this rename is a good idea, because the names `course_org_filter` and `org_filter_out_set` are used elsewhere.

The options are:
1. Leave the code as it was.
2. Go with this new code with renames. (I'm game.)
3. Do a version of this PR with refactor and updated comments, but without renaming these variables. (I'm sort of game, but I need to move on.)
4. Change to the new names everywhere. (I'm not game.) :)

Let me know what you think.  Thanks.

Reply:
I would go with #2. I think this is a great refactor which makes this code much clearer for the reader. I wouldn't expect you to have to boil the ocean to make a substantive improvement like this.